### PR TITLE
Addition of computed fields and a performance profiler (Silk) to the local and development environments

### PR DESCRIPTION
**Pull request by Larspolo**
## Description
Added silk to local and development for easier optimalization development.

After `make setup`, visit localhost:8000/silk to see

### CODE REVIEW

File: requirements/ci.txt [hunk 1]
```diff
@@ -17,3 +17,4 @@ pytest==5.2.0 # Testing library
 pytest-cov==2.7.1 # Pytest
 pytest-django==3.5.1 # Pytest
 pytest-xdist==1.30.0 # Run Pytest in parallel
+django-silk==4.0.1 # Optimazation development
```
Comment (mjvkeulen) [inline requirements/ci.txt]: Spelling and not really what Silk does?

File: src/django/VLE/settings/development.py [hunk 2]
```diff
@@ -28,9 +28,18 @@
 SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
 SECURE_SSL_REDIRECT = True
 
-MIDDLEWARE = ['corsheaders.middleware.CorsMiddleware'] + MIDDLEWARE
```
Comment (mjvkeulen) [inline src/django/VLE/settings/development.py]: I prefer keeping CSP on all environments, and for local (maybe development?) allow inline css for Silk

File: requirements/ci.txt [hunk 3]
```diff
@@ -17,3 +17,4 @@ pytest==5.2.0 # Testing library
 pytest-cov==2.7.1 # Pytest
 pytest-django==3.5.1 # Pytest
 pytest-xdist==1.30.0 # Run Pytest in parallel
+django-silk==4.0.1 # Profiling / Optimisation
```
Comment (mjvkeulen) [inline requirements/ci.txt:20]: Why does the CI env require Silk?

File: src/django/VLE/settings/base.py [hunk 4]
```diff
@@ -125,7 +125,6 @@
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
     'VLE.utils.error_handling.ErrorMiddleware',
-    'csp.middleware.CSPMiddleware',
```
Comment (mjvkeulen) [inline src/django/VLE/settings/base.py:128]: Currently breaks, e.g. on the development env.

**Comment by mjvkeulen** [comment]
Non-local environment require CSP exceptions to handle the inline CSS of Silk

File: requirements/ci.txt (hunk 3 above)
Comment (Larspolo) [reply requirements/ci.txt:20]: Think CI uses local settings

File: src/django/VLE/models.py [hunk 5]
```diff
@@ -1014,34 +1015,65 @@ class Journal(models.Model):
...
+    @computed(models.TextField(null=True), depends=['authors', 'authors#user', 'authors.user#full_name'])
+    def name(self):
+        if self.stored_name:
+            return self.stored_name
+        return ', '.join(self.authors.values_list('user__full_name', flat=True))
 
-    def get_image(self):
-        if self.image:
-            return self.image
+    @computed(models.TextField(null=True), depends=['authors', 'authors#user', 'authors.user#profile_picture'])
```
Comment (Larspolo) [inline src/django/VLE/models.py:1046]: Mostly check if these dependencies are set correctly

File: src/django/VLE/urls.py [hunk 6]
```diff
@@ -49,4 +49,6 @@
     path('update_lti_groups/', lti.update_lti_groups, name='update_lti_groups'),
 
     path('names/<int:course_id>/<int:assignment_id>/<int:journal_id>/', common.names, name='names'),
+
+    url(r'^silk/', include('silk.urls', namespace='silk'))
```
Comment (mjvkeulen) [inline src/django/VLE/urls.py:53]: This will not work for all environments

Comment (mjvkeulen) [reply src/django/VLE/urls.py:53]: Bump, should we just make this a conditional registration?

Comment (Larspolo) [reply src/django/VLE/urls.py:53]: Yes, but was prioritizing notfications over this PR, as it is mostly needed for expansion as we discussed

File: .travis.yml [hunk 7]
```diff
@@ -31,7 +31,7 @@ env:
     - BROKER_URL="amqp://ejournal:password@localhost/ejournal"
     - LTI_SECRET="4339900ae5861f3086861ea492772864"
     - LTI_KEY="0cd500938a8e7414ccd31899710c98ce"
-    - DJANGO_SETTINGS_MODULE="VLE.settings.local"
```
Comment (mjvkeulen) [inline .travis.yml:34]: We have now opted to work with an additional settings module dedicated to travis. This configuration differs from local primarily by excluding silk, right? Does this then warrant a full settings file? Does our Travis config differ further?

Comment (Larspolo) [reply .travis.yml:34]: Does this then warrant a full settings file? Think so

Does our Travis config differ further? No

**Comment by engelhamer** [comment]
@Larspolo @mjvkeulen have you decided on this yet?

**Comment by Larspolo** [comment]
Development now also has silk: https://github.com/eJourn-al/eJournal/pull/906/commits/2aba43afeee6b6bc51e899b467641e80e7cefed8

**Pull request merged**